amethyst_ui/src/pass.rs

```diff
@@ -144,13 +146,17 @@ impl Pass for DrawUi {
             ui_transform,
             mut ui_text,
             editing,
+            hidden,
+            hidden_prop,
         ): <Self as PassData>::Data,
     ) {
         // Populate and update the draw order cache.
         {
             let bitset = &mut self.cached_draw_order.cached;
             self.cached_draw_order.cache.retain(|&(_z, entity)| {
-                let keep = ui_transform.contains(entity);
+                let keep = ui_transform.contains(entity)
+                    && !hidden.contains(entity)
+                    && !hidden_prop.contains(entity);
```

Comment:
Did you test using the ui example to ensure you could hide and unhide with success?

Reply:
Yes, I tested every example with `Hidden`, `HiddenPropagate` (with and without the `HideHierarchySystem` enabled).

Plus my own game's menu's, which are UiPrefabs